app/models/authentication.rb

```diff
@@ -125,6 +125,11 @@ def self.build_credential_options
     end
   end
 
+  def native_ref
```

Comment:
Why do we need this?

Reply:
To make it an interface across the inheritance chain